lib/BloqadeQMC/src/ising/Rydberg.jl

```diff
@@ -1,23 +1,23 @@
 using Base.Iterators
 
-
 abstract type AbstractRydberg{O <: AbstractOperatorSampler} <: AbstractLTFIM{O} end
 
-struct Rydberg{O,M <: UpperTriangular{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, L <: Lattice} <: AbstractRydberg{O}
+struct Rydberg{O,M <: AbstractMatrix{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, A} <: AbstractRydberg{O}
     op_sampler::O
     V::M
     Ω::UΩ
     δ::Uδ
-    lattice::L
+    atoms::A
     energy_shift::Float64
 end
-nspins(H::Rydberg) = nspins(H.lattice)
+
+nspins(H::Rydberg) = length(H.atoms)
```

Comment:
there is `nsites` function already, why are we creating a new interface here?

Reply:
Yes, we'll do that in the refactoring PR to not mess up other parts of the code where nspins is called :)